digiwf-task/digiwf-polyflow-connector-starter/src/main/java/io/muenchendigital/digiwf/task/listener/AssignmentAssignTaskListener.java

```diff
@@ -0,0 +1,46 @@
+package io.muenchendigital.digiwf.task.listener;
+
+import io.holunda.polyflow.taskpool.collector.task.TaskEventCollectorService;
+import io.muenchendigital.digiwf.task.TaskManagementProperties;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.camunda.bpm.engine.delegate.DelegateTask;
+import org.springframework.context.event.EventListener;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+import static io.holunda.camunda.bpm.data.CamundaBpmData.writer;
+import static io.muenchendigital.digiwf.task.TaskVariables.TASK_ASSIGNEE;
+
+@Component
+@Slf4j
+@RequiredArgsConstructor
+public class AssignmentAssignTaskListener {
+  private final TaskManagementProperties.AssignmentProperties properties;
+
+  @Order(TaskEventCollectorService.ORDER - 1000) // be before polyflow
+  @EventListener(condition = "#task.eventName.equals('assignment')")
+  public void taskAssigned(final DelegateTask task) {
+
+    if (properties.isShadow()) {
```

Comment:
I'll try to keep the "Return Early" pattern, but I don't think we discussed that within the team. There are no coding guidlines or stuff like that.